app/src/main/java/com/github/freeman/bootcamp/DrawHatActivity.kt

```diff
@@ -0,0 +1,361 @@
+                    redoVisibility.value = redoCount != 0
+                }
+            }
+        }
+    }
+}
+
+
+// Note: Please see the comments for DrawHatScreen
+@Composable
+private fun DrawHatControlsBar(
+    drawController: DrawController,
+    onColorClick: () -> Unit,
+    onWidthClick: () -> Unit,
+    undoVisibility: MutableState<Boolean>,
+    redoVisibility: MutableState<Boolean>,
+    colorValue: MutableState<Color>,
+    onEraseClick: () -> Unit,
+    currentColor: MutableState<Color>,
+    currentWidth: MutableState<Float>,
+) {
+    val context = LocalContext.current
+    val isToggled = remember { mutableStateOf(false) }
+
+    Row(modifier = Modifier.padding(12.dp), horizontalArrangement = Arrangement.SpaceAround) {
+        MenuItems(
+            R.drawable.ic_undo,
+            LocalContext.current.getString(R.string.undo),
+            if (undoVisibility.value) MaterialTheme.colors.primary else MaterialTheme.colors.secondary
+        ) {
+            if (undoVisibility.value) drawController.unDo()
+        }
+        MenuItems(
+            R.drawable.ic_redo,
+            LocalContext.current.getString(R.string.redo),
+            if (redoVisibility.value) MaterialTheme.colors.primary else MaterialTheme.colors.secondary
+        ) {
+            if (redoVisibility.value) drawController.reDo()
+        }
+        MenuItems(
+            R.drawable.ic_color,
+            context.getString(R.string.stroke_color),
+            colorValue.value
+        ) {
+            onColorClick()
+        }
+        MenuItems(
+            R.drawable.ic_width,
+            LocalContext.current.getString(R.string.stroke_width),
+            MaterialTheme.colors.primary
+        ) {
+            onWidthClick()
+        }
+        // Erases by drawing over the image in white
+        ToggleButton(
+            R.drawable.black_eraser,
+            R.drawable.white_eraser,
+            context.getString(R.string.toggleEraser),
+            MaterialTheme.colors.primary,
+            onClick = {
+                // Toggle the eraser on or off
+                if (!isToggled.value) {
+                    drawController.changeColor(Color.White)
+                    drawController.changeStrokeWidth(DEFAULT_ERASE_WIDTH)
+                    isToggled.value = !isToggled.value
+                    onEraseClick()
+                } else {
+                    drawController.changeColor(currentColor.value)
+                    drawController.changeStrokeWidth(currentWidth.value)
+                    isToggled.value = !isToggled.value
+                }
+            },
+            isToggled = isToggled
+        )
+
+        MenuItems(
+            R.drawable.ic_sharp_arrow_circle_up,
+            LocalContext.current.getString(R.string.drawing_done),
+            MaterialTheme.colors.primary
+        ) {
+            drawController.saveBitmap()
+        }
+    }
+}
+
+@Composable
+fun RowScope.ToggleButton(
```

Comment:
Wasn't it possible to use the MenuItems already defined in the DrawingActivity instead of creating this ToggleButton ?
Genuine question, I might be missing something.

Reply:
Good question. That is actually what I did initially, but if you do that you won't be able to know if you are currently drawing and erasing and I think this would be an unacceptable user experience. I tried using menu items but was always confused and frustrated for the eraser as sometimes I would forget that I turned it on and then I would erase instead of draw. When creating apps we are making them first and foremost for users and so they need to be put first.